esmvaltool/preprocessor/_mask.py

```diff
@@ -449,16 +562,45 @@ def mask_fillvalues(products,
 
 
 def _get_fillvalues_mask(cube, threshold_fraction, min_value, time_window):
+    """
+    Compute the per-model missing values mask.
 
+    Construct the mask that fills a certain time window with missing values
+    if the number of values in that specific window is less than a given
+    fractional threshold; it uses a custom iris Aggregator function that
+    aggregates the cube data by a given time window and counts the number of
+    valid (unmasked) data points within that window;
+    a simple value thresholding is also applied if needed.
+
+    Parameters
+    ----------
+
+    * cube (iris.Cube.cube instance):
+        data cube to be masked.
+
+    * threshold_fraction (float, between 0 and 1):
+        fractional threshold of missing (masked) data points.
+
+    * min_value (float):
+        minumum value threshold to be used as argument for Aggregator.
+
+    * time_window (float):
+        time window in which missing (masked) data points are counted.
+
+    Returns
+    -------
+    numpy masked array.
+
+    """
     # basic checks
     if threshold_fraction < 0 or threshold_fraction > 1.0:
         raise ValueError(
             "Fraction of missing values {} should be between 0 and 1.0".format(
                 threshold_fraction))
     nr_time_points = len(cube.coord('time').points)
     if time_window > nr_time_points:
-        logger.warning("Time window (in time units) larger "
-                       "than total time span")
+        msg = "Time window (in time units) larger than total time span. Stop."
+        raise MaskModuleError(msg)
```

Comment:
Just raise a normal ValueError, that is preferred over creating a zoo of exceptions

Reply:
ok but I liked my Masked error :mask: